### PR TITLE
Refactor Black and isort file exclusions into one data structure

### PR DESCRIPTION
**Pull request by akaihola**
This is in preparation for flynt support (#266 & #308).

Based on #376, merge that one before this PR.

### CODE REVIEW

**Comment by akaihola** [comment]
@jedie, here's another refactoring which was done in preparation for flynt support. Could you review this one as well?

File: src/darker/__main__.py [hunk 1]
```diff
@@ -132,17 +131,16 @@ def _isort_and_blacken_single_file(  # pylint: disable=too-many-arguments
...
-    else:
-        rev2_isorted = rev2_content
-    if relative_path_in_rev2 not in black_exclude:
+    rev2_isorted = apply_isort(
+        rev2_content,
+        relative_path_in_rev2,
+        exclude.isort,
+        edited_linenums_differ,
+        black_config.get("config"),
+        black_config.get("line_length"),
```
Comment (jedie) [inline src/darker/__main__.py:140]: Maybe pass the complete `black_config` object and the function can use the values he needs?

File: src/darker/__main__.py [hunk 2]
```diff
@@ -442,7 +440,7 @@ def main(argv: List[str] = None) -> int:
                 path.relative_to(root) for path in files_to_process
             }
         black_exclude = {
-            f for f in changed_files_to_process if root / f not in files_to_blacken
+            str(f) for f in changed_files_to_process if root / f not in files_to_blacken
```
Comment (jedie) [inline src/darker/__main__.py:443]: Maybe rename `f` to `path` ?

File: src/darker/tests/test_import_sorting.py [hunk 3]
```diff
@@ -55,7 +55,43 @@ def test_apply_isort(git_repo, encoding, newline, content, expect):
...
+        content_, src, exclude=[], edited_linenums_differ=edited_linenums_differ
+    )
+
+    assert result.lines == expect
+    assert result.encoding == encoding
+    assert result.newline == newline
+
+
+@pytest.mark.parametrize("encoding", ["utf-8", "iso-8859-1"])
+@pytest.mark.parametrize("newline", ["\n", "\r\n"])
```
Comment (jedie) [inline src/darker/tests/test_import_sorting.py:68]: maybe add `\r` from Mac? `["\n", "\r\n", "\r"]`

File: src/darker/__main__.py (hunk 1 above)
Comment (akaihola) [reply src/darker/__main__.py:140]: I've considered that, but the rest of `black_config` is Black specific configuration, and we're calling the `apply_isort()` function here. I liked to be explicit here.

Maybe an `IsortOptions` dataclass could encapsulate both of these instead.

Anyway, I decided to not touch this right now.

File: src/darker/__main__.py (hunk 2 above)
Comment (akaihola) [reply src/darker/__main__.py:443]: Thanks, I renamed that.

File: src/darker/tests/test_import_sorting.py (hunk 3 above)
Comment (akaihola) [reply src/darker/tests/test_import_sorting.py:68]: I started adding such test cases (there are lots of these across the test suite), and started getting test failures. See the [mac-newline-tests](akaihola/darker/tree/mac-newline-tests) branch.

However, I haven't yet heard of any complaints from macos users about newlines being treated wrong. Are you sure about `\r`?

**Pull request merged**